source/val/validate_extensions.cpp

```diff
@@ -2226,13 +2226,36 @@ spv_result_t ValidateExtInst(ValidationState_t& _, const Instruction* inst) {
             ValidateOperandDebugType(_, "Base Type", inst, 5, ext_inst_name);
         if (validate_base_type != SPV_SUCCESS) return validate_base_type;
         for (uint32_t i = 6; i < num_words; ++i) {
-          CHECK_OPERAND("Component Count", SpvOpConstant, i);
+          bool invalid = false;
           auto* component_count = _.FindDef(inst->word(i));
-          if (!_.IsIntScalarType(component_count->type_id()) ||
-              !component_count->word(3)) {
+          if (component_count->opcode() == SpvOpConstant) {
+            if (!_.IsIntScalarType(component_count->type_id()) ||
+                !component_count->word(3)) {
+              invalid = true;
+            }
+          } else if (OpenCLDebugInfo100Instructions(component_count->word(4)) ==
+                         OpenCLDebugInfo100DebugLocalVariable ||
+                     OpenCLDebugInfo100Instructions(component_count->word(4)) ==
+                         OpenCLDebugInfo100DebugGlobalVariable) {
+            auto* component_count_type = _.FindDef(component_count->word(6));
+            if (OpenCLDebugInfo100Instructions(component_count_type->word(4)) !=
+                    OpenCLDebugInfo100DebugTypeBasic ||
+                (OpenCLDebugInfo100DebugBaseTypeAttributeEncoding(
+                     component_count_type->word(7)) !=
+                     OpenCLDebugInfo100Signed &&
```

Comment:
> Otherwise its type must be
 a <<DebugTypeBasic,*DebugTypeBasic*>> whose 'Size' is 32 or 64 and whose
 'Encoding' is *Unsigned*.

Doesn't seem that Signed encoding is valid. Also the sizes are not checked.

Reply:
done